corehq/apps/hqcase/case_deletion_utils.py

```diff
@@ -0,0 +1,122 @@
+from attrs import define, field
+
+from casexml.apps.case.xform import extract_case_blocks, get_case_updates
+
+from corehq.apps.hqwebapp.doc_info import get_case_url
+from corehq.form_processor.interfaces.processor import CaseUpdateMetadata
+from corehq.form_processor.models import CommCareCase, XFormInstance
+
+
+@define
+class DeleteCase:
+    name = field()
+    url = field()
+    is_primary = field(default=False)
+    delete_forms = field(factory=list)
+
+
+@define
+class DeleteForm:
+    name = field()
+    url = field()
+    is_primary = field(default=False)
+    affected_cases = field(factory=list)
+
+
+@define
+class FormAffectedCases:
+    case_name = field(default=None)
+    is_current_case = field(default=False)
+    actions = field(factory=list)
+
+
+@define
+class AffectedCase:
+    id = field()
+    name = field()
+    url = field()
+    affected_forms = field(factory=list)
+
+
+def get_affected_case(domain, case, affected_cases_display):
+    for affected_case in affected_cases_display:
+        if affected_case.id == case.case_id:
+            return affected_case
+    if not case.name:
+        case.name = get_deleted_case_name(case)
+    affected_case = AffectedCase(id=case.case_id, name=case.name, url=get_case_url(domain, case.case_id))
+    affected_cases_display.append(affected_case)
+    return affected_case
+
+
+@define
+class AffectedForm:
+    name = field()
+    url = field()
+    actions = field()
+    is_primary = field(default=False)
+
+
+@define
+class ReopenedCase:
+    name = field()
+    url = field()
+    closing_form_url = field()
+    closing_form_name = field()
+    closing_form_is_primary = field(default=False)
+
+
+def get_ordered_case_xforms(case, domain):
+    # Returns deduplicated and chronologically ordered case xforms, if not already that
+    revoked_inclusive_xforms = [t.form_id for t in case.transactions if t.is_form_transaction]
+    xform_objs = [XFormInstance.objects.get_form(form_id, domain) for form_id in revoked_inclusive_xforms]
```

Comment:
going to change this to to `= list({XFormInstance.objects.get_form(form_id, domain) for form_id in revoked_inclusive_xforms})` since it's going to be ordered by the received_on date anyway.